src/components/qcResultsUpload/QcResultsUploadForm.vue

```diff
@@ -71,18 +72,27 @@ export default {
       usedBySelected: 'extraction',
       busy: null,
       disableUpload: null,
+      uploadSuccessful: null,
     }
   },
   computed: {
     qcResultUploadsRequest: ({ api }) => api.traction.qc_results_uploads.create,
     border() {
-      if (this.file === null) return 'border-0'
-      else {
+      if (this.uploadSuccessful === true) {
```

Comment:
you could just have this.uploadSuccessful as the true is implicit. You could also use Also use the ternary operator i.e. `return this.uploadSuccessful ? 'rounded border border-green-500' : 'rounded border border-red-500'` as anything that is not true should be false e.g. null.

Reply:
Yes I was wondering if there was a cleaner way!
The only thing I can think, is that here false = red, and null = no colour. So they have different behaviour, so maybe we need them all?

Reply:
I initially thought the same that it can be replaced with a conditional operator, but then saw it can represent three different states null, true and false not null = false in this case. That;s why I thought a comment would be better ,so it will be clear.

Reply:
I've added a comment but will have a play to see if it can be refactored. Thanks both

Reply:
I think if we enable --strictNullChecks in typescript config file (that we usually do) , it will complain and we need to explicitly use boolean|null as the type for the variable.
But , the checking will work I think because we are  specifically checking it against false (this.uploadSuccesful == false) and not using if(!uploadSuccesful)

Reply:
Its cool that you have the typescript knowledge to predict this in advance, to help with the transition over later! 👏

Reply:
Not an expert, just enough to keep up the work. More to explore

Reply:
@stevieing @seenanair So is the current implementation of this okay do we think? If so, I'll deploy to UAT today

Reply:
It looks fine to me 👍

Reply:
Happy for it go but just worth discussing at some point if it is worth using a type. We use it effectively in lighthouse ui https://github.com/sanger/lighthouse-ui/blob/develop/components/StatusAlert.vue. In this case we are using null true and false to indicate 3 different states so to remove ambiguity we could use an enum which would give a more explicit idea of what is going on. Maybe something worth doing if and when we attempt to move to typescript ...

Reply:
Also, I have come across a similar use in ONTFlowCell with three states. So, if it is a type maybe it can re-used